pathvalidate/_base.py

```diff
@@ -48,15 +59,25 @@ def __init__(
 
         self.__platform = normalize_platform(platform)
 
-        if platform_max_len is None:
-            platform_max_len = self._get_default_max_path_len()
+        # determine max filepath length
+        if platform_max_filepath_len is None:
+            platform_max_filepath_len = self._get_default_max_path_len()
 
-        if max_len <= 0:
-            self._max_len = platform_max_len
+        if max_filepath_len is None or max_filepath_len <= 0:
+            self._max_filepath_len = platform_max_filepath_len
         else:
-            self._max_len = max_len
+            self._max_filepath_len = min(max_filepath_len, platform_max_filepath_len)
```

Comment:
Something I noticed while writing this is that the current implementation effectively ignores any max_len (or max_filepath_len) value which is higher than the platform's max path length (thus putting a hard limit on max_filepath_len of 4096). I think it would be better not to do this, but changing it would be a breaking change.